### ISSUE TITLE
Not able to use predict.py on videos

### ISSUE THREAD

**Issue by HariP1234** [experiencor/basic-yolo-keras#206]
Hi,
I implemented the project in Python 3.5. I trained using 100 images for 50 epochs and the detection is fine when performed on images. But when I use videos it shows error.

python predict.py -c config.json -w my_new_weights.h5 -i my_video.MKV

cv2.error: C:\projects\opencv-python\opencv\modules\imgproc\src\resize.cpp:4044: error: (-215) ssize.width > 0 && ssize.height > 0 in function cv::resize

Video and weight file are in the root directory so I know path is correct. Does someone else have this issue? Am I doing anything wrong?

**Comment by experiencor** [comment]
Hi @HariP1234! It's only work wtih MP4 video. I remember that you need to install OpenCV from source, not from pip or conda, for cv2 to work properly with video.

**Comment by HariP1234** [comment]
Hi **experiencor**,
Sorry for the late reply. I did as you suggested. Downloaded and installed opencv from sourceforge.net but the error didn't go away. I am new to object detection, so maybe I did something wrong. Anyway I decided to use Yolo Step-by-Step process. 
It worked fine and the explanation and comments really helped me, thank you. It converted the video file as well. Detection was not that accurate. That is probably because I didn't add enough images for training.  I do have some doubts.

Epoch 8/100
4/4 [==============================] - ETA: 3:37 - loss: 0.577 - ETA: 2:25 - loss: 0.603 - ETA: 1:12 - loss: 0.547 - 318s 79s/step - loss: 0.4877 - val_loss: 0.2303
Epoch 00008: val_loss did not improve
Epoch 00008: early stopping

1. What is the difference between loss and val_loss? Which one should be minimum?
2. During my first phase of training I used 'yolo.weights' and generated my own weight file. The first training stopped early after 8 epochs. Can the new weight file be used again to train the dataset instead of 'yolo.weights' in Yolo Step-by-Step? Will that increase accuracy?
3. How many images are preferred for training and good accuracy for a new class and to prevent early stopping?
4. I converted a 24 seconds mp4 video of size 842kb of low resolution. It takes about 16 minutes to convert that. I am doing this in my laptop with 4GB nvidia 960MX graphics. Is this insufficient for fast performance? Is there any way to speed up the process and what resolution of video is preferred?

I am a newbie and really need some guidance. Please  help me.

**Comment by experiencor** [comment]
@HariP1234 

1. loss is on on train set and val_loss is on validation. We want both to be small but the later is used in early stopping.

2. Yes. No problem with that. You have a better start and the training is going to be shorter but accuracy remains the same.

3. Early stopping is expected, not to be prevented. 200 images are enough based on my experience.

4. You need to process the images in batches to make it faster. My code processes one image at a time.

**Comment by HariP1234** [comment]
This clarifies a lot of things. Thank you so much.
